File: nysol_python/mcmd/methods/convTex/db/smp/mrand.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='mrand'

############################### IDATA
db['idatas']=[]

data={}
data['name']='dat1.csv'
data['text']='''
customer
A
B
C
D
E
'''
db['idatas'].append(data)

data={}
data['name']='dat2.csv'
data['text']='''
customer
A
A
A
B
B
C
D
D
D
'''
db['idatas'].append(data)

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
0.0から1.0の範囲の実数乱数を生成する。
'''
script['sh_code']='''
mrand a=rand i=dat1.csv o=rsl1.csv
'''
script['py_code']='''
nm.mrand(a="rand", i="dat1.csv", o="rsl1.csv").run()
'''
script['odatas']='rsl1.csv'
db['scripts'].append(script)

script={}
script['title']='基本例2'
script['comment']='''
-intで整数乱数
'''
script['sh_code']='''
mrand a=rand -int i=dat1.csv o=rsl2.csv
'''
script['py_code']='''
nm.mrand(a="rand", int=True, i="dat1.csv", o="rsl2.csv").run()
'''
script['odatas']='rsl2.csv'
db['scripts'].append(script)

script={}
script['title']='最小値、最大値を決めた乱数の生成'
script['comment']='''
最小値が10、最大値が100の整数の乱数を生成し、 ``rand`` という項目名で出力する。
'''
script['sh_code']='''
mrand a=rand -int min=10 max=100 S=1 i=dat1.csv o=rsl3.csv
'''
script['py_code']='''
nm.mrand(a="rand", int=True, min="10", max="100", S="1", i="dat1.csv", o="rsl3.csv").run()
'''
script['odatas']='rsl3.csv'
db['scripts'].append(script)

script={}
script['title']='キー単位の乱数生成'
script['comment']='''
以下の例は、顧客 ``A,B,C,D`` の4人について同じ顧客には同じ乱数値を振る。
'''
script['sh_code']='''
mrand k=顧客 -int min=0 max=1 a=rand i=dat2.csv o=rsl4.csv
'''
script['py_code']='''
nm.mrand(k="customer", int=True, min="0", max="1", a="rand", i="dat2.csv", o="rsl4.csv").run()
'''
script['odatas']='rsl4.csv'
db['scripts'].append(script)

File: olddoc/mcmd/en/examples/mrand.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

File.open("dat1.csv","w"){|fpw| fpw.write(
<<'EOF'
Customer
A
B
C
D
E
EOF
)}

File.open("dat2.csv","w"){|fpw| fpw.write(
<<'EOF'
Customer
A
A
A
B
B
C
D
D
D
EOF
)}


############## Example 1
title="Basic example"
comment=<<'EOF'
Generate random real numbers between 0.0 to 1.0.
EOF
scp=<<'EOF'
more dat1.csv
mrand a=rand i=dat1.csv o=rsl1.csv
more rsl1.csv
EOF
run(scp,title,comment)

############## Example 2
title="Basic Example 2"
comment=<<'EOF'
Generate random integers with -int.
EOF
scp=<<'EOF'
mrand a=rand -int i=dat1.csv o=rsl2.csv
more rsl2.csv
EOF
run(scp,title,comment)

############## Example 3
title="Specify the minimum and maximum value of the random number"
comment=<<'EOF'
Generate a random number with a minimum value of 10 and maximum value of 100. Add the random numbers to a new column named \verb|rand|.
EOF
scp=<<'EOF'
mrand a=rand -int min=10 max=100 S=1 i=dat1.csv o=rsl3.csv
more rsl3.csv
EOF
run(scp,title,comment)
############## Example 4
title="Generate random number by key"
comment=<<'EOF'
Given 4 customers \verb|A,B,C,D|, same random number is generated for same customer.
EOF
scp=<<'EOF'
more dat2.csv
mrand k=Customer -int min=0 max=1 a=rand i=dat2.csv o=rsl4.csv
more rsl4.csv
EOF
run(scp,title,comment)
